src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -0,0 +1,209 @@
+  linksData?: SocialMediaLinkDataProps[];
+  /** Optional true/false to display names of platform along with icon
+   *  NOTE: Labels will NOT be shown with a circular border */
+  showLabels?: boolean;
+  /** Optional size: medium, large, xlarge */
+  size?: SizeType;
+}
+
+/* Accepts an array containing one or more data objects, each representing a social media platform and optional
+ * overrides for the label and/or URL.
+ *
+ * @returns an array of data objects for each requested platform type that includes type, iconName, labelText and
+ * url. The labelText and url props will include any values supplied to override the defaults.
+ */
+function getLinksData(platforms: SocialMediaLinkDataProps[]) {
+  let allData = [];
+  platforms.forEach((myPlatform) => {
+    // Get the dataset for this platform.
+    let thisPlatformArray = socialMediaDataMap.filter(
+      (socialMediaSlice) => socialMediaSlice.type === myPlatform.type
+    );
+
+    // The filter function returns an array of objects, so let's just get the objects out of the array.
+    let thisPlatformData = thisPlatformArray[0];
+
+    // If a url value exists in myPlatform use it, else use the default.
+    let newUrl = myPlatform.url ?? thisPlatformData.url;
+
+    // If a labelText value exists in myPlatform use it, else use the default.
+    let newLabelText = myPlatform.labelText ?? thisPlatformData.labelText;
+
+    let thisObj = {
+      type: thisPlatformData.type,
+      iconName: thisPlatformData.iconName,
+      labelText: newLabelText,
+      url: newUrl,
+    };
+    allData.push(thisObj);
+  }); // end foreach
+
+  return allData;
+}
+
+/* Helper function for .mdx <Table... /> component.
+ * It is located in SocialMediaLinks.tsx because I cannot figure out how to create a function in the .mdx file
+ *
+ * @returns the SocialMediaDataMap values as an array of arrays
+ */
+export function GetTableData() {
+  let tableData = [];
+  socialMediaDataMap.forEach((smPlatform) => {
+    let row = [
+      smPlatform.type,
+      smPlatform.labelText,
+      smPlatform.url,
+      smPlatform.iconName,
+    ];
+    tableData.push(row);
+  });
+  return tableData;
+}
+
+/**
+ * The SocialMediaLinks component renders a list of links for accessing social media sites.
+ */
+export const SocialMediaLinks = chakra(
+  // @todo This throws a JS error b/c it expects a ref parameter. But if I add it, I get a TS error
+  //   because I don't know what to do with the ref, and it says, "unused." The List component doesn't like
+  //   receiving the ref prop and throws a different TS error.
+  forwardRef<HTMLDivElement, SocialMediaLinksProps>((props) => {
+    const {
+      borders = "none",
+      color = "textDefault",
+      className,
+      id,
+      linksData,
+      layout = "row",
+      showLabels = false,
+      size = "small",
+      ...rest
+    } = props;
+
+    // Turns out you can pass whatever props you want to this thing in order to do logic in the theme.
```

Comment:
Ha, this is generally true but also note that you have to define or extract those props in the associated theme file function.